api/src/main/java/org/ccci/idm/user/dao/UserDao.java

```diff
@@ -129,6 +130,16 @@
     List<User> findAllByGroup(@Nonnull Group group, boolean includeDeactivated) throws DaoException;
 
     /**
+     * Find the user with the specified Designation
+     *
+     * @param designation        designation being searched for.
+     * @param includeDeactivated If <tt>true</tt> then deactivated accounts are included.
+     * @return Requested {@link User} or <tt>null</tt> if not found.
+     */
+    @Nullable
```

Comment:
Any reason why `findByDesignation` and `designation` are `@Nullable` whereas `findByEmployeeId` is not?